### DISCUSSION TITLE
rest api  by kerberos

### DISCUSSION THREAD

**Discussion by llh898901** [apache/kyuubi#6851] [Q&A]
kyuubi.authentication = KERBEROS
how to use rest api create batches 
when i set headers authorization=Basic *** or NEGOTIATE *** ,that error 401 no anth scheme matched for ****
which  account can use to access api

**Comment by pan3793** [comment]
you should enable SPNEGO authN and use a client that supports SPNEGO(i.e. `curl` 7.38.0 or later) to access that.
```
kyuubi.spnego.keytab
kyuubi.spnego.principal
```

**Comment by Madhukar525722** [comment]
Hi @pan3793 @turboFei @llh898901 
I am having a kerberized environment, where i want to expose rest api endpoint.  But I am getting error
The endpoints are accessable when authentication is set to NONE.
```
HTTP ERROR 401 No auth scheme matched for null
2025-03-03 11:21:37.116 INFO KyuubiRestFrontendService-100 org.apache.kyuubi.server.http.authentication.AuthenticationAuditLogger: user=null(auth:null) ip=XX proxyIp=null    method=POST     uri=/api/v1/sessions    params=null     protocol=HTTP/1.1       status=401
```

The configs I have used are:
kyuubi.authentication=KERBEROS
kyuubi.frontend.bind.host=0.0.0.0
kyuubi.frontend.bind.port=10009
kyuubi.frontend.rest.bind.host=0.0.0.0
kyuubi.frontend.rest.bind.port=10099
kyuubi.engine.security.enabled=true
kyuubi.engine.security.secret.provider=simple
kyuubi.engine.security.secret.provider.simple.secret=Kyuubi123@
kyuubi.spnego.keytab=spnego.keytab
kyuubi.spnego.principal=HTTP/_HOST@DOMAIN.COM
kyuubi.kinit.principal=hive/_HOST@DOMAIN.COM
kyuubi.kinit.keytab=hive.keytab

```
curl --version
curl 7.61.1 (x86_64-redhat-linux-gnu) libcurl/7.61.1 OpenSSL/1.1.1k zlib/1.2.11 brotli/1.0.6 libidn2/2.2.0 libpsl/0.20.2 (+libidn2/2.2.0) libssh/0.9.6/openssl/zlib nghttp2/1.33.0
Release-Date: 2018-09-05
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz brotli TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
```
